src/Yoti.Auth/ShareUrl/Policy/DynamicPolicy.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Yoti.Auth.ShareUrl.Policy
+{
+    /// <summary>
+    /// Set of data required to request a sharing transaction
+    /// </summary>
+    public class DynamicPolicy
+    {
+        [JsonProperty(PropertyName = "wanted")]
+        private readonly ICollection<WantedAttribute> _wantedAttributes;
+
+        [JsonProperty(PropertyName = "wanted_auth_types")]
+        private readonly List<int> _wantedAuthTypes;
+
+        [JsonProperty(PropertyName = "wanted_remember_me")]
+        private readonly bool _wantedRememberMeId;
+
+        [JsonProperty(PropertyName = "wanted_remember_me_optional")]
+        private readonly bool _isWantedRememberMeIdOptional;
+
+        public DynamicPolicy(
+            ICollection<WantedAttribute> wantedAttributes,
+            List<int> wantedAuthTypes,
+            bool wantedRememberMeId,
+            bool isWantedRememberMeIdOptional)
+        {
+            _wantedAttributes = wantedAttributes;
+            _wantedAuthTypes = wantedAuthTypes;
+            _wantedRememberMeId = wantedRememberMeId;
+            _isWantedRememberMeIdOptional = isWantedRememberMeIdOptional;
+        }
+
+        /// <summary>
+        /// Set of required <see cref="WantedAttribute"/>s
+        /// </summary>
+        public ICollection<WantedAttribute> WantedAttributes
```

Comment:
Perhaps we should make these getters private (or remove them) as we only need to build and serialise to JSON? (Same would apply to other SDKs)